File: c++/sort_algorithms/insertion_sort.cpp
#include <iostream>
using namespace std;

void insertion_sort(int l[], int n)
{
    int pivot = l[0]; 
    int j;

    for (int i=0; i < n; i++)
    {
        pivot = l[i];
        j = i - 1;
        while ( (j >= 0) && (l[j] > pivot) )
        {
            l[j+1] = l[j];
            j--;
        }   
        l[j+1] = pivot;
    } 
}

File: c++/sort_algorithms/insertion_sort.py
def insertion_sort(l:list) -> list:
    """Insertion sort algorithm
        
        Args:
            l (list): list of integers
        
        Return:
            list: the sorted list
        
        Time Complexity:
            O(n²)
    """
    n = len(l)
    for i in range(1, n):
        pivot = l[i]
        j = i - 1 
        while j >= 0 and l[j] > pivot:
            l[j+1] = l[j]
            j -= 1
        l[j+1] = pivot
